enzy_htp/structure/structure_region/capping.py

```diff
@@ -50,124 +62,222 @@ def capping_with_residue_terminals(raw_region,
                       " composed by whole residues only.")
         raise ValueError
 
-    free_res_mapper = raw_region.involved_residues_with_free_terminal()
-    free_nter_res = free_res_mapper["n_ter"]
-    free_cter_res = free_res_mapper["c_ter"]
-    
-    add_atoms = []
-    for res in free_nter_res:
-        cap_atoms = cap_residue_free_terminal(res, nterm_cap, "nterm")
-        add_atoms.extend(cap_atoms)
-
-    for res in free_cter_res:
-        cap_atoms = cap_residue_free_terminal(res, cterm_cap, "cterm")
-        add_atoms.extend(cap_atoms)
-    
+    new_atoms:List[Atom] = list()
+    for res in raw_region.involved_residues:
+        
+        if raw_region.needs_nterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, nterm_cap, "nterm"))
+
+        new_atoms.extend(res.atoms)
+
+        if raw_region.needs_cterm_cap(res):
+            new_atoms.extend(cap_residue_free_terminal(res, cterm_cap, "cterm"))
+
     if return_copy:
         new_region = raw_region.clone()
-        new_region.atoms.extend(add_atoms)
+        new_region.atoms = new_atoms 
         return new_region
     else:
-        raw_region.atoms.extend(add_atoms)
-
-# endregion
+        raw_region.atoms = new_atoms 
```

Comment:
I guess they works the same but the API from StructureRegion is different. Do you mind if we change it back?

Reply:
I would prefer it stay this way. I make the chance because the atom ordering was getting messed up and would make it so that constraints would not work. It seems silly (and probably is) but when done this way, constraints are applied to the correct atoms.

Reply:
I will change this back.

Reply:
Changed back.